### DISCUSSION TITLE
[seeking advice / best practice] should projen generated files be excluded/ignored by code formatters / linters?

### DISCUSSION THREAD

**Discussion by janeklb** [projen/projen#2962] [Q&A] [answered]
For example: if my project has prettier enabled, should prettier be configured to _ignore_ all of the files generated by projen? If yes, is there a way to have projen automatically generate the list of files it manages and add them to the relevant ignore file (`.prettierignore`)?
If no, should/could projen format code as part of `synth-workspace`?

There probably isn't a one-size-fits-all answer, but im curious to know if there's a general direction people lean towards. I'm also new with projen so am not sure if it's even meant to be opinionated about this kind of thing or not.

**Comment by mrgrain** [comment, marked as answer]
Hi @janeklb That's an excellent questions! I think at the moment the recommendation would be to not auto-format and lint any projen generated files, simply because this will fail for most files given that files are generally write-protected.

In future we might want a solution that explicitly allows formatters and linters to hook into a file before it's "locked".

**Reply by janeklb** [reply]
Appreciate the answer @mrgrain - that makes sense. Does projen offer a way to get a list of all of the files it manages (eg. in order to populate `.prettierignore` file)?

**Reply by mrgrain** [reply]
Not directly, but you could go through the `project.components` array and find all components that are a `FileBase`.
